### PR TITLE
EMP-474-fe-hide-empty-crm-section

### PR DESCRIPTION
**Pull request by BNmoj**
changes -

- amended crmDisplayHelper.ts
- amended crm14LegalRepresentation.njk
- amended crmDisplayService.ts
- amended crmDisplayService.test.ts
- amended crmEvidence.njk

### CODE REVIEW

File: server/services/crmDisplayService.ts [hunk 1]
```diff
@@ -57,17 +57,15 @@ export default class CrmDisplayService {
     }
   }
 
-  getSections<T extends CrmResponse>(crmType: CrmType, sectionId: string, crmResponse: T): Array<Section> {
+  getSections<T extends CrmResponse>(crmType: CrmType, sectionId: string, crmResponse: T): Section[] {
```
Comment (muyinatech) [inline server/services/crmDisplayService.ts:60]: Can we keep this as Array, to keep it consistent will usage throughout the codebase.
```suggestion
  getSections<T extends CrmResponse>(crmType: CrmType, sectionId: string, crmResponse: T): Array<Section> {
```

File: server/services/crmDisplayService.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export default class CrmDisplayService {
     const crmDisplayConfig = this.getCrmDisplayConfig(crmType)
 
     let isAnySectionActive = false
-    const items: Array<NavigationItem> = crmDisplayConfig.sections
+    const items: NavigationItem[] = crmDisplayConfig.sections
```
Comment (muyinatech) [inline server/services/crmDisplayService.ts:36]:
```suggestion
    const items: Array<NavigationItem> = crmDisplayConfig.sections
```

File: server/utils/crmDisplayHelper.ts [hunk 3]
```diff
@@ -1,12 +1,75 @@
...
+// Utility to check if a section is empty
+const isSectionEmpty = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
+  if (!section.subsections || section.subsections.length === 0) {
+    return false
+  }
+
+  return section.subsections.every(subsection => isSubsectionEmpty(subsection, crmResponse))
+}
+
+export const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:28]: Only used in this file, so export can be removed
```suggestion
const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
```

File: server/utils/crmDisplayHelper.ts [hunk 4]
```diff
@@ -1,12 +1,75 @@
...
+    const customDisplayValue = getApiFieldValue(crmResponse, subsection.customDisplay.apiField)
+    return !fieldHasValue(customDisplayValue)
+  }
+
+  // Default to empty if no fields or custom display
+  return true
+}
+
+// Utility to determine if a value is "empty"
+export const fieldHasValue = (value: unknown): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:55]: Only used in this file, so export can be removed
```suggestion
const fieldHasValue = (value: unknown): boolean => {
```

File: server/utils/crmDisplayHelper.ts [hunk 5]
```diff
@@ -1,12 +1,75 @@
...
 
 export const includeSection = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
   if (section.hideWhen && conditionsMet(section.hideWhen, crmResponse)) {
     return false
   }
-  return !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  const include = !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  if (include && isSectionEmpty(section, crmResponse)) {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:12]: Add a test case for when section is empty under the   describe('includeSection()' ... in crmDisplayHelper.test.ts

File: server/utils/crmDisplayHelper.ts [hunk 6]
```diff
@@ -1,12 +1,75 @@
...
+const conditionsMet = <T extends CrmResponse>(conditions: Array<ShowOrHideWhen>, crmResponse: T): boolean => {
+  return conditions.every(condition => conditionIsTrue(condition, crmResponse))
+}
+
+const conditionIsTrue = <T extends CrmResponse>(condition: ShowOrHideWhen, crmResponse: T): boolean => {
+  const apiFieldValue = getApiFieldValue(crmResponse, condition.apiField)
+  return condition.equals === String(apiFieldValue)
 }
 
 export const getApiFieldValue = <T extends CrmResponse>(crmResponse: T, apiFieldName: string): string => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:75]: Can` getApiFieldValue` be moved back to after` includeSection`, so we have all the exported functions at the top of the file.

**Review by muyinatech — CHANGES_REQUESTED** [review]

File: server/services/crmDisplayService.test.ts [hunk 7]
```diff
@@ -116,49 +80,7 @@ describe('CRM Display Service', () => {
...
-          { href: '/crm5/1234567/office-use-only', text: 'Office Use Only', active: false },
-          { href: '/crm5/1234567/evidence', text: 'Evidence', active: false },
-          { href: '/crm5/1234567/further-information', text: 'Further Information', active: false },
-          { href: '/crm5/1234567/summary', text: 'Summary', active: false },
-        ],
-        label: 'Side navigation',
-      })
-    })
-
-    it('should return crm navigation for conditions met', () => {
```
Comment (muyinatech) [inline server/services/crmDisplayService.test.ts:134]: This test case needs to be reinstated as it was previously, as it tests the showWhen condition.

File: server/services/crmDisplayService.test.ts [hunk 8]
```diff
@@ -203,73 +125,76 @@ describe('CRM Display Service', () => {
...
-      expect(result).toEqual([
-        {
-          sectionId: 'court-of-appeal-funding',
-          subsections: [{ fields: [], title: 'Court of Appeal Funding' }],
-          title: 'Court of Appeal Funding',
-        },
-      ])
-    })
-
-    it('should return crm section for conditions met', () => {
```
Comment (muyinatech) [inline server/services/crmDisplayService.test.ts:243]: This test case needs to be reinstated - 'should return crm section for conditions met'

File: server/services/crmDisplayService.test.ts [hunk 9]
```diff
@@ -203,73 +125,76 @@ describe('CRM Display Service', () => {
...
-
-      expect(result.length).toEqual(1)
-      expect(result[0].sectionId).toEqual('capital-details')
-      expect(result[0].title).toEqual('Capital Details')
+      expect(result).not.toContainEqual(expect.objectContaining({ sectionId: 'court-of-appeal-funding' }))
     })
 
     it('should return first section if conditions not met for given section', () => {
-      const result = crmDisplayService.getSections('crm5', 'capital-details', crm5Response)
+      const result = crmDisplayService.getSections('crm5', 'non-existent-section', crm5Response)
```
Comment (muyinatech) [inline server/services/crmDisplayService.test.ts:189]: Can wen change this back:-
```suggestion
      const result = crmDisplayService.getSections('crm5', 'capital-details', crm5Response)
```

File: server/utils/crmDisplayHelper.ts (hunk 4 above)
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:55]: Can this function be moved to `utils.ts` and a test cases added.

File: server/utils/crmDisplayHelper.ts [hunk 10]
```diff
@@ -20,6 +21,45 @@ export const getApiFieldValue = <T extends CrmResponse>(crmResponse: T, apiField
   return apiFieldValue as string
 }
 
+export const isSectionEmpty = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:24]: This is not used anywhere, please remove.

File: server/utils/crmDisplayHelper.ts [hunk 11]
```diff
@@ -20,6 +21,45 @@ export const getApiFieldValue = <T extends CrmResponse>(crmResponse: T, apiField
...
+  if (subsection.customDisplay) {
+    const customDisplayValue = getApiFieldValue(crmResponse, subsection.customDisplay.apiField)
+    return !fieldHasValue(customDisplayValue)
+  }
+
+  // Default to empty if no fields or custom display
+  return true
+}
+
+export const shouldIncludeInNavigation = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:55]: You can remove export only used in file

File: server/utils/crmDisplayHelper.ts [hunk 12]
```diff
@@ -20,6 +21,45 @@ export const getApiFieldValue = <T extends CrmResponse>(crmResponse: T, apiField
   return apiFieldValue as string
 }
 
+export const isSectionEmpty = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
+  return section.subsections.every(subsection => isSubsectionEmpty(subsection, crmResponse))
+}
+
+// Utility to check if a subsection is empty
+export const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:29]: isSubSectionEmpty is exported, so a unit test case is required.

File: server/utils/crmDisplayHelper.ts [hunk 13]
```diff
@@ -1,6 +1,7 @@
 import { CrmResponse } from '@eqApi'
-import { Section, ShowOrHideWhen } from '@crmDisplay'
+import { Section, ShowOrHideWhen, Subsection } from '@crmDisplay'
 import _ from 'lodash'
+import { fieldHasValue } from './utils'
 
 export const includeSection = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
```
Comment (muyinatech) [inline server/utils/crmDisplayHelper.ts:6]: You can remove export, only used in file and can moved to the bottom. The unit test case in crmDisplayHelper.test.ts can be removed.

**Review by muyinatech — APPROVED** [review]
